app/services/concerns/xccdf/hosts.rb

```diff
@@ -9,20 +9,40 @@ def save_host
       @host.update!(name: inventory_host['fqdn'])
     end
 
-    def save_profile_host
-      test_result_profile.clone_to(account: @account, host: @host)
+    def host_profile
+      @host_profile ||= test_result_profile.clone_to(
+        account: @account, host: @host
+      )
     end
+    alias save_profile_host host_profile
 
     def report_host
-      @metadata&.dig('fqdn') || @test_result_file.host
+      @metadata&.dig('fqdn') || @test_result_file.test_result.host
+    end
+
+    def associate_rules_from_rule_results
+      ::ProfileRule.import!(
+        ::Rule.where(
```

Comment:
As far as I can tell, this is going to attempt to relate rules with profiles when the rules are already in our DB.

So the following 2 scenarios are not right if I understood correctly:

1 - Report uploaded, we find a canonical profile for it. This would still attempt to add more rules to the profile, but we don't need that as we have the canonical profile (it would be just a waste of time)
2 - Report uploaded, we don't find a canonical profile for it. The profile is not just a tailored profile but something completely custom - no rules are imported and we import an empty profile. 

I think both could be fixed by checking if the result from `test_result_profile` is a initialized object or not, perhaps even a helper for that would be helpful.

Reply:
Note that no rules are created here at all - we're only creating `ProfileRule` records in the DB. Rule creation only ever happens during the benchmark parse - this doesn't change for tailored profiles.

It turns out `ProfileRule` records are unique by profile_id and rule_id, so it's impossible to associate a profile to a rule twice. This method is run for non-customized test result profiles, but it's essentially a no-op (yes, it wastes some time, but premature optimization and all that...); however, if a rule result exists for this test result that isn't actually associated with the named canonical profile (an inconsistency in the generated report for some reason, likely never to happen), then those rules would be associated with this profile.